x-pack/plugin/searchable-snapshots/src/main/java/org/elasticsearch/index/store/SearchableSnapshotDirectory.java

```diff
@@ -317,6 +318,14 @@ public CacheFile getCacheFile(CacheKey cacheKey, long fileLength) throws Excepti
         return cacheService.get(cacheKey, fileLength, cacheDir);
     }
 
+    public Executor executor() {
+        return threadPool.executor(SearchableSnapshotsConstants.SEARCHABLE_SNAPSHOTS_THREAD_POOL_NAME);
+    }
+
+    public Executor directExecutor() {
```

Comment:
Could also be named `prewarmExecutor()`

Reply:
> Can we name the other executor() method something more specific too?

I'm terrible at naming things (sorry, kids)... what do you think of ` searchExecutor()` ? `cacheWritingExecutor()` ?

Reply:
Tricky one TBH. How about `cacheFetchAsyncExecutor()`?

Reply:
+1